pkg/tfbridge/schema.go

```diff
@@ -503,8 +503,8 @@ func (ctx *conversionContext) makeTerraformInput(
 				}
 			}
 			var psflds map[string]*SchemaInfo
-			if ps != nil {
-				psflds = ps.Fields
+			if ps != nil && ps.Elem != nil {
```

Comment:
Doesn't this regress the behaviour for non-list/map types?

Reply:
This branch is only for nested object types. It corrects the behavior for non-list/map types.

Reply:
We do the same traversal when combining `shim` and `*Info` into the Pulumi schema here:

https://github.com/pulumi/pulumi-terraform-bridge/blob/0064b7bef0aef249510a9869f8e78ff0beee505f/pkg/tfgen/generate.go#L347-L385

Note that we pass `elemInfo` into `makeObjectPropertyType`, not `info.Fields`. `makeObjectPropertyType` later accesses `.Fields` (`.Elem.Fields` from the top level property`):

https://github.com/pulumi/pulumi-terraform-bridge/blob/0064b7bef0aef249510a9869f8e78ff0beee505f/pkg/tfgen/generate.go#L449-L452